pkg/server/controller.go

```diff
@@ -310,6 +319,19 @@ func (ctrl *Controller) isAuthRequired() bool {
 	return ctrl.config.Auth.Google.Enabled || ctrl.config.Auth.Github.Enabled || ctrl.config.Auth.Gitlab.Enabled
 }
 
+func (ctrl *Controller) redirectPreservingBaseURL(w http.ResponseWriter, r *http.Request, urlStr string, status int) {
+	if ctrl.config.BaseURL != "" {
```

Comment:
@petethepig you are already doing that in Constructor of Controller New()?

Reply:
yeah, but I need to modify the url so I'm parsing it again. it's cheap and prevents any potential issues related to reusing an object (url object in this case)